framework/src/util/SharedLibrary.cpp

```diff
@@ -47,16 +47,15 @@
 
 namespace cppmicroservices {
 
-class SharedLibraryPrivate : public SharedData
+class SharedLibraryPrivate
 {
 public:
   SharedLibraryPrivate()
-    : 
-     m_Suffix(US_LIB_EXT)
+    : m_Suffix(US_LIB_EXT)
     , m_Prefix(US_LIB_PREFIX)
   {}
 
-  void* m_Handle{nullptr};
+  std::shared_ptr<void *> m_Handle{ nullptr };
```

Comment:
If this is a raw pointer, then calling load would have no affect on another object if it is aliased with the one which is now opened (necessitating a list of all objects which are aliased with the current one). It's like this because an argument was made for using std::unique_ptr.

Reply:
I agree that the overall design should be discussed, but by making the pointer to the SharedLibraryPrivate objects a unique pointer, I firmly believe that keeping the handle as a std::shared_ptr is okay. More specifically, I think it is alright since it doesn't necessitate the extra call to Load() like your suggestion does. If another SharedLibrary object is created through the copy constructor, all relevant data is copied over and the handles are shared; there is no need to call Load() once in the first object and then again in the second during construction. Additionally, since it is a shared pointer, opening/closing the library from one of the objects will affect the other.